### PR TITLE
silx.gui.colors, silx.gui.plot: Added indexed color names in `rgba` and `PlotWidget`

### LINKED ISSUE #3755: Support an index as color in `PlotWidget.addCurve`
Since there is a list of default colors in `PlotWidget`, the idea is to leverage it by allowing to select colors from it in applications.

First raised here: https://gitlab.esrf.fr/bliss/bliss/-/issues/3665

Alternative: Makes sure there is an API for getting/setting the list of default colors and use it like:
```python
plot = PlotWidget()
plot.addCurve(x, y, color=plot.getDefaultColors(1))
```

Note: This is provided through PR #3835

### PR DESCRIPTION
**Pull request by t20100**
~~! Merge PR #3835 first!~~

This PR should close #3755.
It adds support for `color{index}` name in both `colors.rgba` function and `PlotWidget.addCurve|Histogram`.

To do so, `rgba` takes an extra optional `colors` argument

It all sounds a bit complex to me (e.g., to use `PlotWidget.getDefaultColors()` at some place and use `silx.config.DEFAULT_PLOT_CURVE_COLORS` everywhere else). So need to check how this is going to be used and maybe consider adding a `silx.config.DEFAULT_COLORS`.

Maybe that's better as a `PlotWidget.addCurve|Histogram` only feature at first...

### CODE REVIEW

File: src/silx/gui/colors.py [hunk 1]
```diff
@@ -24,13 +24,21 @@
 """This module provides API to manage colors.
 """
 
+from __future__ import annotations
```
Comment (t20100) [inline src/silx/gui/colors.py:27]: This enable using `tuple`, `dict` or `collections.abc.Sequence` as types with Python 3.7.
See [PEP585](https://peps.python.org/pep-0585/#implementation)

**Review by vallsv — APPROVED** [review]

**Comment by vallsv** [comment]
You could allocate the matcher in the module scope, to make it faster, i think

File: src/silx/gui/plot/PlotWidget.py [hunk 2]
```diff
@@ -1168,7 +1168,10 @@ def addCurve(self, x, y, legend=None, info=None,
             # Override previous/default values with provided ones
             curve.setInfo(info)
             if color is not None:
-                curve.setColor(color)
+                curve.setColor(
+                    colors.rgba(color, colors=self.getDefaultColors())
+                    if isinstance(color, str) else color
+                )
```
Comment (t20100) [inline src/silx/gui/plot/PlotWidget.py:1174]: Only convert `str` since `color` can also be an array of colors, one for each point..

**Comment by t20100** [comment]
Fixed issue with passing an array of colors to `addCurve`/`addHistogram`

**Pull request merged**